HTTPieCodeGenerator.coffee

```diff
@@ -111,8 +111,51 @@ HTTPieCodeGenerator = ->
                 s = ""
                 for key, value of object
                     sign = "#{if typeof(value) == 'string' then "=" else ":="}"
-                    s += "    #{addslashes key}#{sign}#{@json_body_object(value)} \\\n"
+                    if typeof value == 'object'
+                      s += addslashes(key) + sign
+                      if value.length != null
+                        s += @nested_array_object(value, 1)
+                    else
+                      s += '\'{\n' + @nested_hash_object(value, 1) + '}\''
         return s
+        
+    @nested_array_object = (array, level) ->
```

Comment:
I remember why. Besides not realizing I could use that trick, I also did it because the level rules were different. Instead of nesting each level for both...

```json
{
  "list": [
    1,
    2
  ]
}
```

I did this:

```json
{
  "list": [1, 2]
}
```

What do you think? I don't care either way and it's your plugin.